### PR TITLE
remove `cask/cmd/fetch`

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
remove `cask/cmd/fetch` because of https://github.com/Homebrew/brew/issues/14799

### CODE REVIEW

File: Library/Homebrew/test/cask/cmd/fetch_spec.rb [hunk 1]
```diff
@@ -1,60 +0,0 @@
...
-    expect(transmission_location).not_to exist
-    expect(caffeine_location).not_to exist
-
-    described_class.run("local-transmission", "local-caffeine")
-
-    expect(transmission_location).to exist
-    expect(caffeine_location).to exist
-  end
-
-  it "prevents double fetch (without nuking existing installation)" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/cmd/fetch_spec.rb:32]: We don't actually have a test for this and the one below for `brew fetch`. Would be good to preserve these two.

File: Library/Homebrew/test/cmd/fetch_spec.rb [hunk 2]
```diff
@@ -14,4 +18,27 @@
     expect(HOMEBREW_CACHE/"testball--0.1.tbz").to be_a_symlink
     expect(HOMEBREW_CACHE/"testball--0.1.tbz").to exist
   end
+
+  it "prevents double fetch (without nuking existing installation)", :integration_test, :needs_macos do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/cmd/fetch_spec.rb:22]: Any way to do this without an integration test? They are really slow so we tend to only have one per command.

CC @reitermarkus for thoughts

Comment (reitermarkus) [reply Library/Homebrew/test/cmd/fetch_spec.rb:22]: There's `Homebrew.fetch_cask` which takes a `Cask::Download`.

Comment (reitermarkus) [reply Library/Homebrew/test/cmd/fetch_spec.rb:22]: Yes.

File: Library/Homebrew/test/cmd/fetch_spec.rb [hunk 3]
```diff
@@ -4,6 +4,10 @@
 require "cmd/shared_examples/args_parse"
 
 describe "brew fetch" do
+  let(:local_transmission) do
+    Cask::CaskLoader.load(cask_path("local-transmission"))
+  end
+
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/cmd/fetch_spec.rb:10]:
```suggestion
```

File: Library/Homebrew/test/cmd/fetch_spec.rb [hunk 4]
```diff
@@ -14,4 +18,29 @@
...
+    brew "fetch", "testball"
+
+    old_ctime = File.stat(HOMEBREW_CACHE/"testball--0.1.tbz").ctime
+    sleep(1)
+
+    expect { brew "fetch", "testball", "--force", "--no-quarantine" }.to be_a_success
+    new_ctime = File.stat(HOMEBREW_CACHE/"testball--0.1.tbz").ctime
+
+    expect(new_ctime.to_i).to be > old_ctime.to_i
+  end
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/cmd/fetch_spec.rb:45]:
```suggestion
```

**Comment by MikeMcQuaid** [comment]
@hyuraku @reitermarkus Let's do without these test changes for now. Happy to see them in a follow-up PR if they can be made non-integration tests.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**